assemble.py

```diff
@@ -265,6 +273,52 @@ def fetch_restorable_apps(target: FactoryClient.Target, dst_dir: str, shortlist:
     return AppsDesc(apps_fetcher.target_dir(target.name), apps_fetcher.get_target_apps_size(target))
 
 
+def check_and_get_fetched_apps_uri(target: FactoryClient.Target, shortlist: str = None):
+    fetched_apps = None
+    fetched_apps_uri = None
+    if "fetched-apps" in target.json.get("custom", {}):
+        fetched_apps_str = target.json["custom"]["fetched-apps"].get("shortlist")
+        if fetched_apps_str:
+            fetched_apps = set(
+                x.strip() for x in fetched_apps_str.split(',') if x)
+        else:
+            # if `shortlist` is not defined or empty then all target apps were fetched
+            fetched_apps = set(app_name for app_name, _ in target.apps())
```

Comment:
i'm guessing target.apps() is returning a dict. Could probably do this like:
```
fetched_apps = set(target.apps().keys())
```

Reply:
Yep, looks like a slightly better option.